spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -67,7 +67,7 @@ public boolean caseSensitive() {
   }
 
   public boolean localityEnabled() {
-    if (table.io() instanceof HadoopFileIO) {
+    if (Util.isHDFSLocation(table.io(), table.location())) {
       HadoopInputFile file = (HadoopInputFile) table.io().newInputFile(table.location());
```

Comment:
This bit seems repetitive in Spark and Flink. What about adding a new method to `Util`?

```
public static boolean mayHaveBlockLocations(FileIO io, String location) {
  if (usesHadoopFileIO(io, location)) {
    ...
    return LOCALITY_WHITELIST_FS.contains(scheme);
  }

  return false;
}
```

I feel it would align well with other methods in `Util` like `blockLocations`.